contracts/Marketplace1155.sol

```diff
@@ -0,0 +1,161 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/Counters.sol";
+import "github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC1155/ERC1155.sol";
+import "github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/security/ReentrancyGuard.sol";
+import "github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC1155/utils/ERC1155Receiver.sol";
+
+contract marketPlace1155 is ReentrancyGuard, ERC1155Receiver {
+    using Counters for Counters.Counter;
+    Counters.Counter private _itemIds;
+    Counters.Counter private _itemsSold;
+
+    address public owner;
+
+    constructor() {
+        owner = msg.sender;
+    }
+
+    struct MarketItem {
+        uint256 itemId;
+        address nftContract;
+        uint256 tokenIds; // togli s finale
+        address payable seller;
+        address payable owner;
+        uint256 price;
+        bool sold; //serve?
+    }
+
+    mapping(uint256 => MarketItem) private idToMarketItem; // a cosa mi serve la mappa. Sto rimappando i dati di 1155. Se non aggiunge informazione a cosa serve?
+
+    event MarketItemCreated(
+        uint256 indexed itemId,
+        address indexed nftContract,
+        uint256 indexed tokenIds,
+        address seller,
+        address owner,
+        uint256 price,
+        bool sold
```

Comment:
Do we need this `bool`? The check for the unsold item is made with `owner` field. I would remove this piece of redundant information